plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -65,86 +64,112 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
             if (domain.isNullAllowed()) {
                 break;
             }
+
+            int currentlyProcessedClusteringColumn = allProcesedClusteringColumns;
```

Comment:
seems like we could have just one variable?

Reply:
It's this way as I'm using this in lambdas atm and I was getting error due to not final field used in lambda. It's used to detect whether we can do the legal IN statement in Cassandra